.github/workflows/release.yml

```diff
@@ -1,26 +1,126 @@
 name: Release
 
 on:
-  push:
-    tags:
-      - '[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+    inputs:
+      releaseType:
+        type: choice
+        options:
+          - minor
+          - patch
+        default: minor
+        required: true
+        description: 'minor: v0.X.0, patch: v0.0.X'
+      debug:
+        type: boolean
+        default: true
+        description: 'executes the workflow in debug mode (skip the publishing tag, docker image and release steps)'
 
 jobs:
-  release:
-    name: Create Release
-    if: github.event.base_ref == 'refs/heads/master'
+  check-permission:
+    name: Check permission
+    if: contains(github.ref, 'refs/heads/master')
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
     steps:
-      - name: Get Version
-        id: get_version
+      - name: Check user permission
+        uses: actions-cool/check-user-permission@v2.2.0
+        id: check
+        with:
+          require: 'write'
+    outputs:
+      hasWritePermission: ${{ steps.check.outputs.require-result }}
+
+  publish-tag:
+    name: Publish tag
+    needs: check-permission
+    if: contains(needs.check-permission.outputs.hasWritePermission, 'true')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout prebid cache
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Create & publish tag
+        id: release
+        run: |
+          currentTag=$(git describe --abbrev=0 --tags)
+          echo "Current release tag ${currentTag}"
+
+          echo ${currentTag}  | grep -q "[0-9]\+\.[0-9]\+\.[0-9]\+"
```

Comment:
Tag `version0.0.0` or `v.0.0.0` or some variation of these two would still satisfy both `grep -q "[0-9]\+\.[0-9]\+\.[0-9]\+"` and the newly added check `if [[ "${currentTag:0:1}" != "v" ]];`. Can we modify to make sure the format only comes with one leading `v` character? Maybe the regex could become:
```
 -  echo ${currentTag}  | grep -q "[0-9]\+\.[0-9]\+\.[0-9]\+"
 +  echo ${currentTag}  | grep -q "v[0-9]\+\.[0-9]\+\.[0-9]\+"
    if [ $? -ne 0 ]; then
      echo "Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+"
      exit 1
    fi
```

Reply:
@guscarreon  https://github.com/prebid/prebid-cache/pull/135/commits/3d1dec514fb289b72d7eab713f6da8865a430c9b  makes changes to make sure following formats of current release tag are valid:
- `v[major].[minor].[patch]`
- `[major].[minor].[patch]` (Supported now for backward compatibility. Can be removed when new release tag is created with this script)

Testing details:
```
#!/bin/bash

echo $1 | grep -q "^v\?[0-9]\+\.[0-9]\+\.[0-9]\+$"
if [ $? -ne 0 ]; then
  echo "Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+"
  exit 1
fi
echo "valid tag"

Valid cases
------------------------------------------------------------------------------------------------------ 

~ » ./abc.sh v0.1.100
valid tag

~ » ./abc.sh 0.1.100  (Backward compatibility with today's tag)
valid tag

Invalid cases
------------------------------------------------------------------------------------------------------
~ » ./abc.sh 0a.1.100
Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+

~ » ./abc.sh 0.1a.100
Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+

~ » ./abc.sh 0.1.100a
Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+

~ » ./abc.sh version0.1.100
Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+

~ » ./abc.sh a0.1.100
Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+

~ » ./abc.sh ver0.1.100
Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+
```